### ISSUE TITLE
Crash in BlockEntityGroundStorage.DetermineStorageProperties

### ISSUE THREAD

**Issue by gamenerd2026** [anegostudios/VintageStory-Issues#1111]
Been crashing A LOT A LOT recently with the new version

Not sure what I'm doing... been going down a no steep cave and finding 1 of those hunching gray hostile creachers that spawn everyware at night, and have been choping some trees at the base, it keeps on crashing

Logs:
[client-crash.txt](https://github.com/anegostudios/VintageStory-Issues/files/6778600/client-crash.txt)
[client-debug.txt](https://github.com/anegostudios/VintageStory-Issues/files/6778601/client-debug.txt)
[client-main.txt](https://github.com/anegostudios/VintageStory-Issues/files/6778602/client-main.txt)
[server-audit.txt](https://github.com/anegostudios/VintageStory-Issues/files/6778603/server-audit.txt)

Crash Report:
```
Running on 64 bit Windows with 16 GB RAM 
Version: v1.15.0-rc.1 (Unstable)7/7/2021 11:54:38 AM: Critical error occurred
System.NullReferenceException: Object reference not set to an instance of an object.
   at Vintagestory.GameContent.BlockEntityGroundStorage.DetermineStorageProperties(ItemSlot sourceSlot)
   at Vintagestory.GameContent.BlockEntityGroundStorage.FromTreeAttributes(ITreeAttribute tree, IWorldAccessor worldForResolving)
   at Vintagestory.Client.NoObf.ClientChunk.AddOrUpdateBlockEntityFromPacket(_f47bFFjjZjqsZ07abkCWRRjJjq5 p, ClientMain game)
   at Vintagestory.Client.NoObf.GeneralPacketHandler.HandleBlockEntities(_SPt7iF5LhXmcxMRSPZV1A0ZATGh packet)
   at Vintagestory.Client.NoObf.ClientMain.ExecuteMainThreadTasks(Single deltaTime)
   at _Hhyykp3OIyR3A9gDXvfRYXmIxHM._N2PKmadywKhrSAOGDRbet32cyTB(Single )
   at _GatfFd3kn5xATuUyKYscxn8d0CO._4TEOzZagWNPjzZVqfu2HCwbLFqC(Single )
   at _GatfFd3kn5xATuUyKYscxn8d0CO._x3D81oTSuYvhXUwgP6x9PwySgdc(Single )
   at Vintagestory.Client.NoObf.ClientPlatformWindows.window_RenderFrame(Object sender, FrameEventArgs e)
   at System.EventHandler`1.Invoke(Object sender, TEventArgs e)
   at OpenTK.GameWindow.RaiseRenderFrame(Double elapsed, Double& timestamp)
   at OpenTK.GameWindow.DispatchRenderFrame()
   at OpenTK.GameWindow.Run(Double updates_per_second, Double frames_per_second)
   at _woe4UEAalnRbwPXBTK09ZhdFcWO._5qgAk5LLuTcbcRWXQsq1b3Y8T8H(_arFvNaodTjXO5ObZ7eEgimkezLG , String[] )
   at _ZFpbvq06JtO6suJqmLDa8Hs9YeH._5qgAk5LLuTcbcRWXQsq1b3Y8T8H(ThreadStart )
-------------------------------

Event Log entries containing Vintagestory.exe, the latest 3
==================================
{ TimeGenerated = 7/5/2021 10:47:26 AM, Site = , Source = .NET Runtime, Message = Application: Vintagestory.exe
Framework Version: v4.0.30319
Description: The process was terminated due to an internal error in the .NET Runtime at IP 00007FFFBF8E8CF9 (00007FFFBF700000) with exit code 80131506.
 }
--------------
{ TimeGenerated = 7/2/2021 7:48:38 PM, Site = , Source = Windows Error Reporting, Message = Fault bucket 1601164709659358829, type 5
Event Name: RADAR_PRE_LEAK_64
Response: Not available
Cab Id: 0

Problem signature:
P1: Vintagestory.exe
P2: 1.14.10.0
P3: 10.0.19043.2.0.0
P4: 
P5: 
P6: 
P7: 
P8: 
P9: 
P10: 

Attached files:
\\?\C:\Users\koawa\AppData\Local\Temp\RDRFFDA.tmp\empty.txt
\\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WERFFEB.tmp.WERInternalMetadata.xml
\\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WERFFFC.tmp.xml

These files may be available here:


Analysis symbol: 
Rechecking for solution: 0
Report Id: e4f6a818-4855-4749-b5b1-2197727b5e76
Report Status: 268435456
Hashed bucket: e2510997d6d277c626387ad1f223366d
Cab Guid: 0 }
```

**Comment by gamenerd2026** [comment]
1.14.10.0 Idk why it says that, I'm useing 1.15.0-rc 1

**Comment by gamenerd2026** [comment]
It's unplayable

**Comment by copygirl** [comment]
Is this a new world, or a converted one from 1.14? Do you remember placing anything on the ground or recently making a pit kiln?

**Comment by gamenerd2026** [comment]
converted form 1.15.0-pre.12, pit kiln was created in 1.15.0-pre.12, pit kiln is still lit

**Comment by copygirl** [comment]
As @Tarod7 noticed in #1114, the crash also occurs when interacting with a (broken?) pit kiln:
```
System.NullReferenceException: Object reference not set to an instance of an object.
   at Vintagestory.GameContent.BlockEntityGroundStorage.DetermineStorageProperties(ItemSlot sourceSlot)
   at Vintagestory.GameContent.BlockEntityGroundStorage.OnPlayerInteract(IPlayer player, BlockSelection bs)
   at Vintagestory.Client.NoObf.SystemMouseInWorldInteractions.TryBeginUseBlock(Block selectedBlock, BlockSelection blockSelection)
```

**Comment by Tarod7** [comment]
@copygirl the tooltip said something like empty pit. visually there were no items and there was a layer at the bottom. Hopefully that helps.

**Comment by Yury-k** [comment]
The same crash as @gamenerd2026 described, 1.15 rc1 installed together with 1.14.8, but world generated using 1.15, playing over LAN with one more user.
Loading current save will bring this  "BlockEntityGroundStorage.DetermineStorageProperties" error crash in a minute or so, and yes, we have kiln pit burning that should be ready some time.

> Running on 64 bit Windows with 16 GB RAM 
> Version: v1.15.0-rc.1 (Unstable)2021-07-07 09:07:07 PM: Critical error occurred
> System.NullReferenceException: Object reference not set to an instance of an object.
>    at Vintagestory.GameContent.BlockEntityGroundStorage.DetermineStorageProperties(ItemSlot sourceSlot)
>    at Vintagestory.GameContent.BlockEntityGroundStorage.FromTreeAttributes(ITreeAttribute tree, IWorldAccessor worldForResolving)
>    at Vintagestory.Client.NoObf.ClientChunk.AddOrUpdateBlockEntityFromPacket(_f47bFFjjZjqsZ07abkCWRRjJjq5 p, ClientMain game)
>    at Vintagestory.Client.NoObf.GeneralPacketHandler.HandleBlockEntities(_SPt7iF5LhXmcxMRSPZV1A0ZATGh packet)
>    at Vintagestory.Client.NoObf.ClientMain.ExecuteMainThreadTasks(Single deltaTime)
>    at _Hhyykp3OIyR3A9gDXvfRYXmIxHM._N2PKmadywKhrSAOGDRbet32cyTB(Single )
>    at _GatfFd3kn5xATuUyKYscxn8d0CO._4TEOzZagWNPjzZVqfu2HCwbLFqC(Single )
>    at _GatfFd3kn5xATuUyKYscxn8d0CO._x3D81oTSuYvhXUwgP6x9PwySgdc(Single )
>    at Vintagestory.Client.NoObf.ClientPlatformWindows.window_RenderFrame(Object sender, FrameEventArgs e)
>    at System.EventHandler`1.Invoke(Object sender, TEventArgs e)
>    at OpenTK.GameWindow.RaiseRenderFrame(Double elapsed, Double& timestamp)
>    at OpenTK.GameWindow.DispatchRenderFrame()
>    at OpenTK.GameWindow.Run(Double updates_per_second, Double frames_per_second)
>    at _woe4UEAalnRbwPXBTK09ZhdFcWO._5qgAk5LLuTcbcRWXQsq1b3Y8T8H(_arFvNaodTjXO5ObZ7eEgimkezLG , String[] )
>    at _ZFpbvq06JtO6suJqmLDa8Hs9YeH._5qgAk5LLuTcbcRWXQsq1b3Y8T8H(ThreadStart )
> -------------------------------
[client-debug.txt](https://github.com/anegostudios/VintageStory-Issues/files/6781218/client-debug.txt)
[client-main.txt](https://github.com/anegostudios/VintageStory-Issues/files/6781219/client-main.txt)
[client-crash.txt](https://github.com/anegostudios/VintageStory-Issues/files/6781220/client-crash.txt)

**Comment by phellmes** [comment]
I observed it like this:
If I leave burning pit kilns so they finish while I'm away (a certain amount of chunks?), they don't have the expected items in them when I come back. They're visually empty and show "empty pile" one mouse over. If I right-click on that - crash like described above.
If I stay in close range to the pit kilns while they burn and finish I can retrieve the expected items out of them.
Maybe this helps. And the empty pit kiln thing is not new to rc1 - I experienced that with former preX versions, too.
